components/stepper/script.js

```diff
@@ -13,25 +13,109 @@ const { BaseComponent } = components;
  * Class description
  */
 class Stepper extends BaseComponent {
-    /**
-     * Returns the currently selected item
-     */
-    get value() {
-        return this.items[this.selectedIndex];
-    }
+    // eslint-disable-next-line require-jsdoc
+    static get observedAttributes() { return ['value']; }
+
     /* eslint-disable require-jsdoc */
     constructor() {
         super();
 
         this.itemElements = [];
-        this.items = [];
+        this._items = [];
         this.selectedIndex = 0;
         this.template = template;
 
         this.next = this.next.bind(this);
         this.prev = this.prev.bind(this);
-
         this.init = this.init.bind(this);
+
+        this.stateSchema = { value: { type: ['string'] } };
+        this.state = { value: '' };
+    }
+
+    get value() {
+        return this.state.value;
+    }
+
+    set value(value) {
+        this.setAttribute('value', value);
+    }
+
+    /**
+     * Will set new items array to the stepper component and will verify the array
+     * After that will reset the value of the stepper if it does not exist in the new items array
+     * @param {Array<string>} value
+     */
+    set items(value) {
+        if (!Array.isArray(value)) {
+            console.warn('Can\'t set non array value to gameface-stepper items');
+            return;
+        }
+        if (!value.length) {
+            console.warn('Can\'t set an empty array value to gameface-stepper items');
+            return;
+        }
+
+        const nonStringItems = value.filter(el => typeof el !== 'string');
+        if (nonStringItems.length) {
+            console.warn(`Can't set an array values that are not strings to gameface-stepper items. [${nonStringItems.join(', ')}]`);
+            return;
+        }
+
+        this._items = value;
+
+        // If the value does not exists in the new items array then change it to default one
+        // and warn the user for this change
+        this.selectedIndex = this.getSelectedIndex(this.value);
+        if (this.selectedIndex !== -1) return;
+
+        console.warn(`The current value '${this.value}' is not a member of the items. Will set the first item from the items array as value - '${this.items[0]}'`);

```

Comment:
To me it doesn't seem the warning exactly fits here in this case:
![image](https://github.com/CoherentLabs/GameUIComponents/assets/21058220/4aaffb94-39b8-46b1-b2bc-78a69156e4c3)

Reply:
Well, I added a warning here so you do not get confused if the current value is replaced by another text when you change the options.

I prefer to have a warning but if you think it does not fit well I will remove it.

Reply:
No, don't remove it, I was thinking about the message itself.
But I get what the use case might be 1. you have already values in there 2. change to something totally different 3. message comes and sets the selected to the first value from the new array / items.

It is definitely better to have a warning - leave it.